test/controllers/ContinuedIllnessController.test.ts

```diff
@@ -0,0 +1,60 @@
+import { expect } from 'chai';
+import request from 'supertest';
+import { createApp } from '../ApplicationFactory';
+
+import { Appeal } from 'app/models/Appeal';
+import { ApplicationData } from 'app/models/ApplicationData';
+import { Illness } from 'app/models/Illness';
+import { Reasons } from 'app/models/Reasons';
+import { CONTINUED_ILLNESS_PAGE_URI } from 'app/utils/Paths';
+
+describe('ContinuedIllnessController', () => {
+
+    it('should show hint containing illness start date, and radio buttons for Yes and No on GET', async () => {
+        const  applicationData: Partial<ApplicationData> = {
+            appeal: {
+                penaltyIdentifier: {
+                    companyNumber: 'NI000000'
+                },
+                reasons: {
+                    illness: {
+                        illnessStart: new Date(2020, 4, 1)
```

Comment:
Test simulates date in `Date` format but without custom deserialisation `String` type will be used instead. It is because node-session-handler does simple `JSON.stringify()` to serialise and `JSON.parse` to deserialise.

In other words test simulates scenario that will never happen.

Reply:
Ah. So you're saying we _have_ to store the dates as strings? This really doesn't sound like a good idea from a data design perspective.

Reply:
I do not, I am saying after read from sessions store value will be in string type. Type definitions and actual type might be completely different at runtime, it is developers job to make sure they mach 100%.